File: Advanced Programming Methods/Toy Language Java/src/Model/DataStructures/IDictionary.java
package Model.DataStructures;

import java.util.Collection;
import java.util.Set;

//dictionary generic interface, wrapper
//in use by Symbol Table and File Table
public interface IDictionary<K,V> {
    V put(K key, V value);
    int size();
    boolean containsKey(K key);
    boolean containsValue(V value);
    V get(K key) throws MyException;
    String toString();
    Set<K> keySet();
    Collection values();
    void remove(K key);
}

File: Advanced Programming Methods/Toy Language C#/A2/Model/Data Structures/IDictionary.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace A2.Model.Data_Structures
{

    //dictionary generic interface, wrapper
    //in use by Symbol Table
    public interface IDictionary<K, V>
    {
        V put(K key, V value);
        int size();
        Boolean containsKey(K key);
        Boolean containsValue(V value);
        V get(K key);
        string ToString();
        HashSet<K> keySet();
        HashSet<V> values();
        void remove(K key);
    }


}
